bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,284 @@
+    run rdctl create-profile --output reg --hive hkcu --body "$(complex_json_data)"
+    assert_registry_output
+}
+
+@test 'complains when no input source is specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type
+        assert_failure
+        assert_output --partial "no input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+    done
+}
+
+@test 'complains when multiple input sources are specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type --input some-file.txt -b moose
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type --input some-file.txt --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type --input some-file.txt -b moose --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+        run rdctl create-profile --output $type -b moose --from-settings
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+
+    done
+}
+
+simple_json_data() {
+    echo '{ "kubernetes": {"version": "moose-head" }}'
+}
+
+assert_moose_head_plist_output() {
+    assert_success
+    # Just match a few of the lines near the start and the end of the output.
+    # The unit tests do more comprehensive output checking.
+    assert_output - <<'EOF'
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>version</key>
+      <string>moose-head</string>
+    </dict>
+  </dict>
+</plist>
+EOF
+}
+
+@test 'generates plist output from a command-line argument' {
+    run rdctl create-profile --output plist --body "$(simple_json_data)"
+    assert_moose_head_plist_output
+}
+
+@test 'generates plist output from a file' {
+    run rdctl create-profile --output plist --input <(simple_json_data)
+    assert_moose_head_plist_output
+}
+
+@test 'verify plutil is ok with the generated plist output from input file' {
+    if ! is_macos; then
+        skip
+    fi
+    local PLIST_FILE="$BATS_TEST_TMPDIR/rdctl-create-profile.plist"
+    rdctl create-profile --output plist --input <(simple_json_data) >"$PLIST_FILE"
+    run plutil -s "$PLIST_FILE"
```

Comment:
Don't bother with a temp file:

```suggestion
    run rdctl create-profile --output plist --input <(simple_json_data)
    assert_success
    run plutil -s <<<"$output"
```

Reply:
Exporting the shell function removes the need for separate commands. I can just run `rdctl create-profile ... <(simple_json_data) | plutil -s -`